silkworm/rpc/core/evm_debug.cpp

```diff
@@ -543,15 +549,16 @@ Task<void> DebugExecutor::execute(
 
     const auto chain_config = co_await storage.read_chain_config();
     auto current_executor = co_await boost::asio::this_coro::executor;
+
+    // We must do the execution at the state after the txn identified by transaction_with_block param in the same block
+    // at the state of the block identified by the given block_num, i.e. at the start of the block (block_num)
+    const auto first_txn_num_in_block = co_await tx_.first_txn_num_in_block(block.header.number + 1);
```

Comment:
In recent PRs targeting trace/debug API end-points, there are code portions like this one regarding the transaction-level state access which keep returning again and again. When finished, we should strive for some refactoring, at least give it a try.

Nothing to do here specifically, just a general comment 😄